pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,654 @@
+	// ErrIncompatibleExportType serves as error message when an export kind is provided with an incompatible argument
+	ErrIncompatibleExportType = "Should not specify export kind"
+
+	// progressBarCycle is a const used to store the cycle displayed in the progress bar when downloading the exported volume
+	progressBarCycle = `"[___________________]" "[==>________________]" "[====>______________]" "[======>____________]" "[========>__________]" "[==========>________]" "[============>______]" "[==============>____]" "[================>__]" "[==================>]"`
+)
+
+var (
+	// Flags
+	vm           string
+	snapshot     string
+	pvc          string
+	outputFile   string
+	insecure     bool
+	keepVme      bool
+	shouldCreate bool
+	volumeName   string
+)
+
+type exportFunc func(client kubecli.KubevirtClient, vmeInfo *VMExportInfo) error
+
+type HTTPClientCreator func(*http.Transport, bool) *http.Client
+
+type exportCompleteFunc func(kubecli.KubevirtClient, *VMExportInfo, time.Duration, time.Duration) error
+
+// ExportProcessingComplete is used to store the function to wait for the export object to be ready.
+// Useful for unit tests.
+var ExportProcessingComplete exportCompleteFunc = waitForVirtualMachineExport
+
+type VMExportInfo struct {
+	ShouldCreate bool
+	Insecure     bool
+	KeepVme      bool
+	OutputFile   string
+	VolumeName   string
+	Namespace    string
+	Name         string
+	ExportSource k8sv1.TypedLocalObjectReference
+}
+
+type command struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+var exportFunction exportFunc
+
+var httpClientCreatorFunc HTTPClientCreator
+
+// SetHTTPClientCreator allows overriding the default http client (useful for unit testing)
+func SetHTTPClientCreator(f HTTPClientCreator) {
+	httpClientCreatorFunc = f
+}
+
+// SetDefaultHTTPClientCreator sets the http client creator back to default
+func SetDefaultHTTPClientCreator() {
+	httpClientCreatorFunc = getHTTPClient
+}
+
+func init() {
+	SetDefaultHTTPClientCreator()
+}
+
+// usage provides several valid usage examples of vmexport
+func usage() string {
+	usage := `# Create a VirtualMachineExport to export a volume from a virtual machine:
+	{{ProgramName}} vmexport create vm1-export --vm=vm1
+  
+	# Create a VirtualMachineExport to export a volume from a virtual machine snapshot
+	{{ProgramName}} vmexport create snap1-export --snapshot=snap1
+  
+	# Create a VirtualMachineExport to export a volume from a PVC
+	{{ProgramName}} vmexport create pvc1-export --pvc=pvc1
+  
+	# Delete a VirtualMachineExport resource
+	{{ProgramName}} vmexport delete snap1-export
+  
+	# Download a volume from an already existing VirtualMachineExport (--volume is optional when only one volume is available)
+	{{ProgramName}} vmexport download vm1-export --volume=volume1 --output=disk.img.gz
+  
+	# Create a VirtualMachineExport and download the requested volume from it
+	{{ProgramName}} vmexport download vm1-export --pvc=pvc1 --volume=volume1 --output=disk.img.gz`
```

Comment:
maybe for pvc adding the --volume is not the best example since pvc is only 1 volume?